packages/next-app/src/app/pool/[poolId]/(components)/MetadataAttributesTable.tsx

```diff
@@ -111,14 +117,63 @@ function Row({ data }: { data: PoolMetadataAttribute }) {
   );
 }
 
-export function MetadataAttributesTable({ poolId }: { poolId: string }) {
-  const { metadata } = useContext(PoolMetadataContext);
+export function MetadataAttributesTable({ poolId }: { poolId: `0x${string}` }) {
+  const { metadata, handleSetMetadata, handleSetPool } =
+    useContext(PoolMetadataContext);
+
+  handleSetPool(poolId);
+
+  useEffect(() => handleSetMetadata([]), []);
+
+  const { data: pools } = metadataGql.useMetadataPool({
+    poolId,
+  });
+
+  const metadataPool = pools?.pools[0];
+
+  const [poolMetadataCID, setMetadataCID] = useState("");
+  const debouncedCID = useDebounce(poolMetadataCID, 500);
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await fetch(
+        `https://ipfs.io/ipfs/${metadataPool?.metadataCID}`,
+        {
+          method: "GET",
+        }
+      );
+      const json = await response.json();
+      handleSetMetadata(json);
+    }
+    if (metadataPool?.metadataCID) {
+      fetchData();
+    }
+  }, [metadataPool?.metadataCID]);
```

Comment:
`#style` suggestion: return early

```suggestion
    if (!metadataPool?.metadataCID) return
    
    fetchData();
  }, [metadataPool?.metadataCID]);
```